### PR TITLE
Add Maps & Autocomplete for User Location Selection

### PR DESCRIPTION
**Pull request by lolfuljames**
This PR adds:
 - A map UI that shows current location
 - A input text field to select current location
 - Submit button now shows user location & coordinates

demo: https://jtan-sps-summer20.appspot.com/

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -2,10 +2,92 @@ function getRandomRestaurant() {
...
+    }
+
+    infowindowContent.children['place-icon'].src = place.icon;
+    infowindowContent.children['place-name'].textContent = place.name;
+    infowindowContent.children['place-address'].textContent = address;
+    infowindow.open(map, marker);
+  });
+
+
+  document.getElementById("submit-button").addEventListener("click", function(){
```
Comment (lolfuljames) [inline src/main/webapp/script.js:75]: Planning to add further expansions such as sending the location to the servlet within this function

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -2,10 +2,92 @@ function getRandomRestaurant() {
...
+      return;
+    }
+
+    const curNameElement = document.getElementById('current-name');
+    const curLocElement = document.getElementById('current-loc');
+    curNameElement.innerText = place.name;
+    curLocElement.innerText = place.geometry.location.toString();
+
+    getRandomRestaurant();
+  });
```
Comment (lolfuljames) [inline src/main/webapp/script.js:92]: This shall be the main function where most of our processing will start from

Comment (lolfuljames) [inline src/main/webapp/script.js:92]: Main function where most of the processing can be initiated from in the future

File: src/main/webapp/index.html [hunk 3]
```diff
@@ -2,15 +2,45 @@
...
+    ></script>
   </head>
+
   <body>
     <h1>Welcome to EatGoWhere!</h1>
+    <p id="current-name"></p>
+    <p id="current-loc"></p>
     <p id="restaurant-name"></p>
-    <button onclick="getRandomRestaurant()">Submit</button>
+    <div class="pac-card" id="pac-card">
```
Comment (leonyeo) [inline src/main/webapp/index.html:28]: What does pac stand for? Maybe something like searchbox-card will be better?

**Review by leonyeo — COMMENTED** [review]
Looks good thanks James!

File: src/main/webapp/index.html (hunk 3 above)
Comment (lolfuljames) [reply src/main/webapp/index.html:28]: This is following the places api css classes naming, https://developers.google.com/maps/documentation/javascript/places-autocomplete#places_searchbox

It should mean Places Auto Complete. What do you think?

Comment (leonyeo) [reply src/main/webapp/index.html:28]: I see that's fine then!

**Review by leonyeo — APPROVED** [review]

**Pull request merged**